packages/ra-ui-materialui/src/list/SimpleList.tsx

```diff
@@ -131,7 +131,8 @@ const SimpleList = <RecordType extends Record = Record>(
                                 record={data[id]}
                             >
                                 <ListItem
-                                    button={!!linkType as any}
+                                    // Ensure we don't have li items inside our own LI at L126
+                                    component="div"
```

Comment:
Could you revrt this one? This change removes the hand mouse pointer on list items, and I think that the UX is more important than the DOM validation.